### PR TITLE
Set calendarStartDay in DateText to WP "week starts on" [MAILPOET-3703]

### PR DESCRIPTION
**Pull request by brezocordero**
Update also @types/react-datepicker to latest to include calendarStartDay.

[MAILPOET-3703]

[MAILPOET-3703]: https://mailpoet.atlassian.net/browse/MAILPOET-3703

### CODE REVIEW

**Comment by brezocordero** [comment]
> What do you think about that approach?

I like it. Thanks for all the details. I will work on it.

File: views/newsletters.html [hunk 1]
```diff
@@ -22,7 +22,6 @@
       var mailpoet_current_time = <%= json_encode(current_time) %>;
       var mailpoet_schedule_time_of_day = <%= json_encode(schedule_time_of_day) %>;
       var mailpoet_date_display_format = "<%= wp_date_format() %>";
-      var mailpoet_start_of_week = "<%= wp_start_of_week() %>";
```
Comment (brezocordero) [inline views/newsletters.html:25]: Removed this variable as already included in layout.html with mailpoet_wp_week_starts_on

File: assets/js/src/newsletters/send/date_text.jsx [hunk 2]
```diff
@@ -76,8 +76,8 @@ locale.localize.day = buildLocalizeFn({
   values: dayValues,
   defaultWidth: 'wide',
 });
-locale.options.weekStartsOn = typeof window.mailpoet_start_of_week !== 'undefined'
```
Comment (brezocordero) [inline assets/js/src/newsletters/send/date_text.jsx:79]: Replaced with the value in MailPoet object, for consistency.

**Review by costasovo — APPROVED** [review]
Great job! Thanks!

**Review by veljkho — APPROVED** [review]

**Review by veljkho — APPROVED** [review]

**Pull request merged**